modules/table/src/main/java/org/apache/ignite/internal/table/distributed/StorageUpdateHandler.java

```diff
@@ -157,51 +158,77 @@ public void handleUpdateAll(
     ) {
         indexUpdateHandler.waitIndexes();
 
-        storage.runConsistently(locker -> {
-            int commitTblId = commitPartitionId.tableId();
-            int commitPartId = commitPartitionId.partitionId();
-
-            if (!nullOrEmpty(rowsToUpdate)) {
-                List<RowId> rowIds = new ArrayList<>();
-
-                // Sort IDs to prevent deadlock. Natural UUID order matches RowId order within the same partition.
-                SortedMap<UUID, TimedBinaryRow> sortedRowsToUpdateMap = new TreeMap<>(rowsToUpdate);
-
-                for (Map.Entry<UUID, TimedBinaryRow> entry : sortedRowsToUpdateMap.entrySet()) {
-                    RowId rowId = new RowId(partitionId, entry.getKey());
-                    BinaryRow row = entry.getValue() == null ? null : entry.getValue().binaryRow();
-
-                    locker.lock(rowId);
+        if (nullOrEmpty(rowsToUpdate)) {
+            return;
+        }
+        long batchSize = getBatchSize();
+        BiPredicate<List<Entry<UUID, TimedBinaryRow>>, Entry<UUID, TimedBinaryRow>> isFullByRowCount = (batch, row) ->
+                batch.size() + 1 > batchSize;
+        List<List<Map.Entry<UUID, TimedBinaryRow>>> batches = getTimedBinaryRowBatches(rowsToUpdate, isFullByRowCount);
+        batches.forEach((batch) ->
```

Comment:
I would replace this with a foreach loop: the readability is the same, but we spare a capturing lambda.